coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/PlaybackToSecondary.scala

```diff
@@ -265,6 +265,25 @@ class PlaybackToSecondary[CT, CV](u: PlaybackToSecondary.SuperUniverse[CT, CV],
       retrying[T](actions, filter)
     }
 
+    /**
+      * Serialize resync in the same store group
+      * because resync will cause reads to be unavailable (at least in pg)
+      */
+    def resyncSerially(): Unit = {
+      val sm = u.secondaryManifest
+      try {
+        // resync immediately calls commit before actual work is done.
+        // That allows us to see resync in progress from the resync table.
+        sm.lockResync(datasetId, secondary.storeId, secondary.groupName)
+        resync()
+      } catch {
+        case ex: SQLException =>
+          throw ResyncLaterSecondaryException(ex.getMessage)
```

Comment:
Part of me wonders if it would be wise to only catch the expected SQLException here